packages/fetcher/src/__tests__/index.test.ts

```diff
@@ -1,19 +1,130 @@
+} from "undici";
+import type { Fetcher } from "..";
+
+const [nodeMajor] = process.version
+  .slice(1) // remove 'v'
+  .split(".", 1)
+  .map(Number);
 
 // This "test suite" actually does all its work at compile time.
-function isAFetcher(_fetcher: Fetcher) {}
+describe("implements Fetcher", () => {
+  function isAFetcher(_fetcher: Fetcher) {}
 
-it("node-fetch is a Fetcher", () => {
-  isAFetcher(nodeFetch);
-});
+  it("node-fetch", () => {
+    isAFetcher(nodeFetch);
+  });
 
-it("make-fetch-happen is a Fetcher", () => {
-  isAFetcher(makeFetchHappen);
+  it("make-fetch-happen", () => {
+    isAFetcher(makeFetchHappen);
+  });
+
+  it("undici", () => {
+    isAFetcher(undiciFetch);
+  });
 });
 
-it("undici is a Fetcher", () => {
-  isAFetcher(undiciFetch);
+describe("Fetcher accepts FormData as a body", () => {
+  beforeEach(nockBeforeEach);
+  afterEach(nockAfterEach);
+
+  it("node-fetch", async () => {
+    const formData = new FormData();
+    formData.append("foo", "bar");
+
+    nock("https://example.com")
+      .post("/", (body) => {
+        expect(body).toMatch(
+          `Content-Disposition: form-data; name="foo"\r\n\r\nbar`,
+        );
+        return true;
+      })
+      .reply(200, "ok");
+
+    await nodeFetch("https://example.com/", {
+      method: "POST",
+      body: formData,
+    });
+  });
+
+  it("make-fetch-happen", async () => {
+    const formData = new FormData();
+    formData.append("foo", "bar");
+
+    nock("https://example.com")
+      .post("/", (body) => {
+        expect(body).toMatch(
+          `Content-Disposition: form-data; name="foo"\r\n\r\nbar`,
+        );
+        return true;
+      })
+      .reply(200, "ok");
+
+    await makeFetchHappen("https://example.com/", {
+      method: "POST",
+      body: formData,
+    });
+  });
+
+  (nodeMajor! >= 16 ? it : it.skip)("undici", async () => {
+    // We can't use nock with undici, but undici provides mocking utilities
+    // https://github.com/nock/nock/issues/2183
+    const agent = new MockAgent({ connections: 1 });
+    agent.disableNetConnect();
+    setGlobalDispatcher(agent);
+
+    const client = agent.get("https://example.com");
+
+    client
+      .intercept({
+        path: "/",
+        method: "POST",
+        // This is unexpected. the body is typed as a string but it's the actual
+        // FormData object still. Pretty sure this is a bug.
+        body(body: any) {
+          const bar = body.get("foo");
+          return bar === "bar";
+        },
+      })
+      .reply(200, "ok");
+
+    // undici doesn't support `form-data`'s `FormData` class
+    const formData = new UndiciFormData();
+    formData.append("foo", "bar");
+
+    const result = await undiciFetch("https://example.com/", {
+      method: "POST",
+      body: formData,
+    });
+
+    const t = await result.text();
+    expect(t).toMatchInlineSnapshot(`"ok"`);
+  });
 });
+
+// Ensures an active and clean nock before every test
+function nockBeforeEach() {
```

Comment:
i thought you added these somewhere in a different PR?

Reply:
Maybe just time to extract them into a util package? They don't exist in this repo yet. You're thinking of the usage reporting PR.